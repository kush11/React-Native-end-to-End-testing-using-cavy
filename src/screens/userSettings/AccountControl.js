import React, { Component } from 'react';
import {
  View, StyleSheet, Text, Alert, AsyncStorage,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import RazorpayCheckout from 'react-native-razorpay';
import { LoginManager, AccessToken } from 'react-native-fbsdk';
import { GoogleSignin } from 'react-native-google-signin';
import { connect } from 'react-redux';
import {
  Left, Body, Right, List, ListItem, Toast
} from 'native-base';
import { fontMaker } from '../../components/utility/fonts/FontMaker';
import {
  updateUsername, updateDob, updateUserSocialImage, updateFetchedUrl, resetState
} from '../../store/actions/index';
import { updateCurrentFlow } from '../../store/actions/assessment';
import { BASE_URL, headers } from '../../api/config/Config';


const mapStateToProps = state => ({
  uName: state.User.name,
  uSocialImage: state.User.socialImage,
  fetchedUrl: state.User.fetchedURL,
  gender: state.User.gender
});
const mapDispatchToProps = dispatch => ({

  updatedUserName: name => dispatch(updateUsername(name)),
  updatedFetchedUrl: url => dispatch(updateFetchedUrl(url)),
  updatedUserSocialImage: uri => dispatch(updateUserSocialImage(uri)),
  updatedDob: dob => dispatch(updateDob(dob)),
  updatedCurrentFlow: flow => dispatch(updateCurrentFlow(flow)),
  ResetState: () => dispatch(resetState())
});

export class AccountSetting extends Component {
  logout = async () => {
    const { navigation, ResetState } = this.props;
    try {
      await ResetState();
    } catch (error) {
      Toast.show({
        text: 'Error Occured',
        duration: 2000,
        type: 'default'
      });
    }

    this.signOut();
    const { updatedCurrentFlow } = this.props;
    navigation.navigate('StartPage');
    updatedCurrentFlow('NEW');
  }

  signOut = async () => {
    const { updatedDob } = this.props;
    updatedDob(null);
    try {
      await AsyncStorage.removeItem('userToken');
      await AsyncStorage.removeItem('UserSocialImage');
      if (await GoogleSignin.isSignedIn()) { await GoogleSignin.signOut(); }
    } catch (error) {
      Toast.show({
        text: 'Error Occured',
        duration: 2000,
        type: 'default'
      });
    }
    if (AccessToken.getCurrentAccessToken()) {
      await LoginManager.logOut();
    }
  };

  showAlert = () => {
    Alert.alert(
      'Account Deactivated',
      'Your account will be deleted within 50 days. You can login within this period with your credentials.',
      [
        {
          text: 'Ok',
          onPress: () => {
            this.logout();
          }
        },
      ],
      { cancelable: false }
    );
  }

  deleteAccount = () => {
    const { uName } = this.props;
    Alert.alert(
      'Delete Account',
      'Do you want to delete your Account?',
      [
        {
          text: 'NO',
          onPress: () => { },
          style: 'cancel'
        },
        {
          text: 'YES',
          onPress: () => {
            AsyncStorage.removeItem('enableTouchId');
            fetch(`${BASE_URL}/api/updateUserStatus`, {
              method: 'POST',
              headers,
              body: JSON.stringify({ name: uName })
            }).then(() => {
              this.showAlert();
            });
          }
        },
      ],
      { cancelable: false }
    );
  }

  payAccount = () => {
    const options = {
      description: 'Credits towards consultation',
      image: 'https://i.imgur.com/3g7nmJC.png',
      currency: 'INR',
      key: 'rzp_test_9U0bdWxnAUEpRm',
      amount: '5000',
      name: 'ZUL',
      prefill: {
        email: 'sagar.vivek159@gmail.com',
        contact: '9538346231',
        name: 'Razorpay Software'
      },
      theme: { color: '#F37254' }
    };
    RazorpayCheckout.open(options).then((data) => {
      // handle success
      console.log("data", data);
      alert(`Success: ${data.razorpay_payment_id}`);
    }).catch((error) => {
      // handle failure
      console.log("error", error);
      alert(`Error: ${error.code} | ${error.description}`);
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <List>
          <ListItem onPress={this.deleteAccount}>
            <Left>
              <View style={styles.list}>
                <View style={{ alignItems: 'center', alignSelf: 'center', width: 50 }}>
                  <Icon name="user-times" size={20} color="grey" />
                </View>
                <View><Text style={styles.listText}>Delete Account</Text></View>

              </View>
            </Left>

            <Body />
            <Right>
              <Icon name="angle-right" size={20} color="grey" />
            </Right>
          </ListItem>
          {/* <ListItem onPress={this.payAccount}>
            <Left>
              <View style={styles.list}>
                <View style={{ alignItems: 'center', alignSelf: 'center', width: 50 }}>
                  <Icon name="user-times" size={20} color="grey" />
                </View>
                <View><Text style={styles.listText}>Pay Account</Text></View>

              </View>
            </Left>

            <Body />
            <Right>
              <Icon name="angle-right" size={20} color="grey" />
            </Right>
          </ListItem> */}
        </List>
      </View>
    );
  }
}
const menuItemFontStyle = fontMaker({ family: 'Montserrat', weight: 'Regular' });
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white'
  },
  list: {
    flex: 1,
    flexDirection: 'row',
  },
  listText: {
    fontSize: 16,
    color: 'black',
    paddingLeft: 5,
    width: '100%',
    ...menuItemFontStyle
  },
});
export default connect(mapStateToProps, mapDispatchToProps)(AccountSetting);
