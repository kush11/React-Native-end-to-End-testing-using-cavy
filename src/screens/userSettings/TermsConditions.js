import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { fontMaker } from '../../components/utility/fonts/FontMaker';

const termsConditions = (props) => {
  const { children } = props;
  return (
    <View>
      <Text style={styles.listText}>
        {children}
      </Text>
    </View>
  );
};

const termsFontStyle = fontMaker({ family: 'OpenSans', weight: 'Regular' });
const styles = StyleSheet.create({
  container: { flex: 1 },
  list: { flexDirection: 'row' },
  listText: {
    fontSize: 18,
    textAlign: 'center',
    color: 'black',
    ...termsFontStyle
  },
});
export default termsConditions;
