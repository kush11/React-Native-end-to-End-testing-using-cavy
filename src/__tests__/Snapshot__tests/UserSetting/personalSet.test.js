import 'react-native';
import 'isomorphic-fetch';
import React from 'react';
import Renderer from 'react-test-renderer';
import { PersonalSetting } from '../../../screens/userSettings/PersonalSettings';

it('snapshots', () => {
  const tree = Renderer.create(<PersonalSetting />).toJSON();
  expect(tree).toMatchSnapshot();
});
