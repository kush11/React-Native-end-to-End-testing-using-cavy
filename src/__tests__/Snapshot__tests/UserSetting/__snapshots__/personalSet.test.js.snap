// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        enabled={true}
        keyboardVerticalOffset={0}
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 13,
                "flexDirection": "column",
                "overflow": "hidden",
                "width": 270,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255,255,255)",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "margin": 18,
                },
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                  Object {
                    "alignSelf": "center",
                    "color": "black",
                  },
                ]
              }
            >
              Select Gender
            </Text>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "padding": 10,
                    },
                    undefined,
                    Object {
                      "backgroundColor": null,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      Object {
                        "borderColor": "black",
                        "borderRadius": 10,
                        "borderWidth": 1,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "black",
                        "borderRadius": 5,
                        "height": 10,
                        "width": 10,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "marginLeft": 5,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "black",
                        "fontSize": 16,
                      }
                    }
                  >
                    Male
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "padding": 10,
                    },
                    undefined,
                    null,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      Object {
                        "borderColor": "black",
                        "borderRadius": 10,
                        "borderWidth": 1,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "marginLeft": 5,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "black",
                        "fontSize": 16,
                      }
                    }
                  >
                    Female
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "padding": 10,
                    },
                    undefined,
                    null,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      Object {
                        "borderColor": "black",
                        "borderRadius": 10,
                        "borderWidth": 1,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "marginLeft": 5,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "black",
                        "fontSize": 16,
                      }
                    }
                  >
                    Others
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderTopColor": "#A9ADAE",
                  "borderTopWidth": 0.5,
                  "flexDirection": "row",
                  "height": 46,
                  "justifyContent": "space-between",
                },
                Object {},
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#007ff9",
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Cancel
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#A9ADAE",
                    "height": "100%",
                    "width": 0.5,
                  },
                  Object {},
                ]
              }
            />
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "fontSize": 17,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#007ff9",
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Submit
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
        "marginTop": 10,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 3,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          closeOnRowBeginSwipe={false}
          closeOnRowPress={true}
          closeOnScroll={true}
          directionalDistanceChangeThreshold={2}
          disableLeftSwipe={false}
          disableRightSwipe={false}
          leftOpenValue={0}
          previewFirstRow={false}
          recalculateHiddenLayout={false}
          rightOpenValue={0}
          style={Object {}}
          swipeToOpenPercent={50}
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 16,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 16,
                        }
                      }
                    >
                      Display Name
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": null,
                    "alignSelf": null,
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "color": "grey",
                      "fontFamily": "Montserrat",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 16,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 16,
                        }
                      }
                    >
                      Date of Birth
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": null,
                    "alignSelf": null,
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "color": "grey",
                      "fontFamily": "Montserrat",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                    }
                  }
                >
                  Set D.O.B
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 16,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 16,
                        }
                      }
                    >
                      Gender
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": null,
                    "alignSelf": null,
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "color": "grey",
                      "fontFamily": "Montserrat",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 16,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 16,
                        }
                      }
                    >
                      BMI
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": null,
                    "alignSelf": null,
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "color": "grey",
                      "fontFamily": "Montserrat",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 16,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 16,
                        }
                      }
                    >
                      Height
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": null,
                    "alignSelf": null,
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "color": "grey",
                      "fontFamily": "Montserrat",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                    }
                  }
                >
                  Set height
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "alignSelf": "center",
                    "flex": 0.28,
                    "paddingBottom": null,
                    "paddingLeft": null,
                    "paddingRight": null,
                    "paddingTop": null,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "grey",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 16,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "OpenSans-Regular",
                          "fontSize": 16,
                        }
                      }
                    >
                      Weight
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": null,
                    "alignSelf": null,
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "color": "grey",
                      "fontFamily": "Montserrat",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                    }
                  }
                >
                  Set weight
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "alignSelf": "center",
                    "flex": 0.28,
                    "paddingBottom": null,
                    "paddingLeft": null,
                    "paddingRight": null,
                    "paddingTop": null,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "grey",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.58)",
          "bottom": 0,
          "flex": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flex": 1,
                "margin": 16,
                "paddingBottom": 16,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 8,
                  "marginRight": 8,
                  "marginTop": 24,
                }
              }
            >
              <View>
                <RCTDatePicker
                  date={1549862953594}
                  locale="en_EN"
                  mode="date"
                  onChange={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "height": 216,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={Object {}}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "#007AFF",
                          "fontSize": 16,
                          "textAlign": "center",
                        }
                      }
                    >
                      Cancel
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={Object {}}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "#007AFF",
                          "fontSize": 16,
                          "textAlign": "center",
                        }
                      }
                    >
                      Ok
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
